microsoft-edge/progressive-web-apps-chromium/experimental-features/index.md

```diff
@@ -0,0 +1,469 @@
+The window controls overlay ensures users can still minimize, maximize or restore, and close the app. It also provides access to relevant browser controls via the web app menu. For Chromium based browsers the controls in the overlay are:  
+
+*   A draggable region the same width and height of each of the window control buttons  
+*   the "Settings and more" three-dot button  
+*   the window control buttons minimize, maximize, restore and close.  
+    
+There can be scenarios where the browser will show other content in  the controls overlay.  
+
+*   When an installed web app is launched, the origin of the webpage will display to the left of the three-dot menu for a few seconds, then disappear  
+*   If a user interacts with an extension via the **Settings and more** menu, the icon of the extension will appear in the overlay to the left of the three-dot menu.  After you exit any extension dialog, the icon is removed from the overlay.  
+    
+For right to left based languages, the overlay is displayed in the upper right corner of the client area. For left to right based languages, the controls is flipped and is displayed on the upper left of the client area.
+
+>[!IMPORTANT]
+> The overlay will always be on top of the web content's Z order and will accept all user input without flowing it through to the web content.  
+
+### Working Around the Window Controls Overlay
+
+Your web content will need to be aware of the reserved area for the controls overlay and ensure that area isn't expecting user interaction. To accomplish this, you will need to query the browser for the bounding rectangle and visibility of the controls overlay.  This information is provided to you through JavaScript APIs and CSS environment variables.
+
+**JavaScript APIs**
+
+A new object on the `window.navigator` property called `windowControlsOverlay` has been introduced to query the bounding rectangle of the controls overlay.
+
+`windowControlsOverlay` has two objects:  
+
+*   `getBoundingClientRect()` which returns a `DOMRect` representing the area under the window controls overlay.  
+*   `visible` a boolean to determine if the controls overlay has been rendered.  
+    
+> [!IMPORTANT]
+> For privacy reasons, the `windowControlsOverlay` will not be accessible to iframes within the web content
+
+Whenever the overlay is resized, a `geometrychange` event will be fired on the `navigator.windowControlsOverlay` object to notify the client that it should recalculate the content layout based on the new bounding rectangle of the overlay. 
+
+**CSS Environment Variables**
+
+In addition to the JavaScript API above, the bounding rectangle of the controls overlay can also be queried using CSS. Four new CSS environment variables have been introduced to accomplish this:  
+
+*   `titlebar-area-inset-top`  
```

Comment:
These might change to `titlebar-area-x/y/width/height` before 91. Note that it drops the word "inset".

Reply:
Ok thanks for the heads up. Let's see where we land next week.

Reply:
The change has signoff upstream and is just running through CI now, so we should use the new variables :)
https://chromium-review.googlesource.com/c/chromium/src/+/2780946/